### DISCUSSION TITLE
Some questions about the CLI

### DISCUSSION THREAD

**Discussion by matsn0w** [vercel/vercel#7172] [Help]
Hey there,

I am trying to set up a DevOps workflow through GitLab CI for a large amount of projects. I've developed a generic pipeline for my deployments which can be reused across multiple projects. Each projects consists of a Laravel API and a Nuxt.js frontend. Both live in a separate folder within the repo.

I am deploying the Laravel API part to a 'normal' web host, and now I'm trying to deploy the Nuxt part to Vercel using the CLI. Both are deployed at the same time, and must connect to each other. This is done by using the (short) commit hash as unique identifier in domain names. The `API_URL` variable points to the correct domain using this hash.

This way, you can get the following situation:
- The Nuxt app lives at `<commit hash>.review.domain.com`
- The API lives at `api.<commit hash>.review.domain.com`

It was quite difficult to set the custom domains on the Vercel side using the CLI, but I got it to work.

Now I am struggling to get a few other basic things working that in my opinion should be easy. Maybe I am overtinking stuff.

1. How to get the ID of a deployment? As said, at runtime I only 'know' the commit hash as unique identifier.
2. How to deploy as 'CI runner'? I think using a personal token is a bad practice. Should I create a dedicated account for this?

I might stumble across some more questions, but these are my main problems now.

Thanks for your thoughts and help in advance!

**Comment by nickspaargaren** [comment]
The `vercel deploy` command output returns the generated preview environment, it's possible to read this value.

Take a look at: https://docs.gitlab.com/ee/ci/environments/index.html#set-dynamic-environment-urls-after-a-job-finishes
